### PR TITLE
JSUI-3048 Hid facet search results when selecting or excluding a value

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/JSUI-3048

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: unitTests/ui/ValueElementTest.ts [hunk 1]
```diff
@@ -42,6 +43,8 @@ export function ValueElementTest() {
       facet = Mock.optionsComponentSetup<Facet, IFacetOptions>(Facet, {
         field: '@field'
       }).cmp;
+      facet.ensureDom();
+      facet.facetSearch = Mock.mock(FacetSearch);
```
Comment (btaillon) [inline unitTests/ui/ValueElementTest.ts:47]: To remember in the future: if `ensureDom` isn't called before setting the mock, the mock may get overwritten later...

**Review by ThibodeauJF — APPROVED** [review]

File: src/ui/Facet/ValueElement.ts [hunk 2]
```diff
@@ -198,6 +198,12 @@ export class ValueElement {
     eventArg.clear();
   }
 
+  private tryDismissSearchResults() {
+    if (this.facet && this.facet.facetSearch && this.facet.facetSearch.dismissSearchResults) {
+      this.facet.facetSearch.dismissSearchResults();
+    }
```
Comment (samisayegh) [inline src/ui/Facet/ValueElement.ts:204]: I feel this approach is not very encapsulated, where we are trying to adjust the behaviour of the facet search from the outside.

Would it work if we instead added a listener inside the facet search, such that when a select/exclude is detected, it hides the search results? Is there an event we can hook onto?

File: unitTests/ui/ValueElementTest.ts [hunk 3]
```diff
@@ -118,9 +121,44 @@ export function ValueElementTest() {
...
+        expect(dismissSearchResults).toHaveBeenCalledTimes(1);
+      });
+
+      it('when the click event on the exclude icon is triggered, dismisses search results', () => {
+        const { dismissSearchResults } = facet.facetSearch;
+        expect(dismissSearchResults).not.toHaveBeenCalled();
+        mockExcludeIcon.element.click();
+        expect(dismissSearchResults).toHaveBeenCalledTimes(1);
+      });
     });
```
Comment (samisayegh) [inline unitTests/ui/ValueElementTest.ts:162]: I feel this `describe` block and the two `it` blocks below are repetitive tests that can be removed. Unless I am missing something, I think the tests on lines 128 and 135 are enough.

Comment (btaillon) [reply unitTests/ui/ValueElementTest.ts:162]: Normally, they should be enough, but `select` and `exclude` go through different functions, which caused only one of those to properly dismiss search results.

File: src/ui/Facet/ValueElement.ts (hunk 2 above)
Comment (btaillon) [reply src/ui/Facet/ValueElement.ts:204]: There is no existing event, but I can add one.

**Pull request merged**